src/AutoRest.CSharp/Common/Input/CodeModelConverter.cs

```diff
@@ -404,5 +404,93 @@ public static IReadOnlyList<string> GetApiVersions(CodeModel codeModel)
                 .Distinct()
                 .OrderBy(v => v)
                 .ToList();
+
+        private static InputAuth GetAuth(IEnumerable<SecurityScheme> securitySchemes)
+        {
+            InputApiKeyAuth? apiKey = null;
+            InputOAuth2Auth? oAuth2 = null;
+
+            foreach (var scheme in securitySchemes.Distinct(SecuritySchemesComparer.Instance))
+            {
+                switch (scheme)
+                {
+                    case AzureKeySecurityScheme:
+                        throw new NotSupportedException($"{typeof(AzureKeySecurityScheme)} is not supported. Use {typeof(KeySecurityScheme)} instead");
+                    case AADTokenSecurityScheme:
+                        throw new NotSupportedException($"{typeof(AADTokenSecurityScheme)} is not supported. Use {typeof(OAuth2SecurityScheme)} instead");
+                    case KeySecurityScheme when apiKey is not null:
```

Comment:
This might not be necessary, since in `SecuritySchemesComparer` it's ensured that only one `.Type` can exist.

Reply:
True. Which means that it is implemented incorrectly. Nice catch!